packages/slate-editor/src/modules/editor/plugins/withDeserializeHtml/createDeserialize/createMarksDeserializer.ts

```diff
@@ -1,26 +1,21 @@
-import type { Extension } from '@prezly/slate-commons';
+import type { DeserializeMarks } from '@prezly/slate-commons';
 import type { Descendant } from 'slate';
 import { Element } from 'slate';
 import { createText } from 'slate-hyperscript';
 
-import { getLeafDeserializers } from './getLeafDeserializers';
-
 type DeserializeHTMLChildren = ChildNode | Descendant | string | null;
 
-export function deserializeHtmlToMarks(extensions: Extension[]) {
-    return function (node: HTMLElement, children: DeserializeHTMLChildren[]): Descendant[] | null {
+export type MarksDeserializer = (
+    node: HTMLElement,
+    children: DeserializeHTMLChildren[],
+) => Descendant[] | null;
+
+export function createMarksDeserializer(deserializers: DeserializeMarks): MarksDeserializer {
+    return function (node, children) {
         const type = node.getAttribute('data-slate-type') || node.nodeName;
-        const deserializers = getLeafDeserializers(extensions);
```

Comment:
Same here. This single-use calculation was lifted one level up to allow reusing it for a deserialization run.